### PR TITLE
refactor: migrate queries into the graphql folder: Assets

### PR DESCRIPTION
**Pull request by theodu**

### CODE REVIEW

**Comment by theodu** [comment]
- create the AssetQuery function
- do not move the client-facing function assets in internal functions
- refacto the `_copy_assets`method in `copy_project` to use the batch loop already implemented in graphQLquery  (rather define a post call function that download and upload assets when calling AssetQuery)
- adapt tests

**Comment by theodu** [comment]
I will run E2E tests to be more confident: https://github.com/kili-technology/kili-python-sdk/actions/runs/3930936061

**Review by Jonas1312 — APPROVED** [review]

File: src/kili/graphql/__init__.py [hunk 1]
```diff
@@ -70,24 +71,40 @@ def query(fragment: str) -> str:
...
+    FRAGMENT_TYPE: Type = NotImplemented
+
+    # pylint: disable=useless-return
+    def get_post_call_function(self, post_call_options: Optional[NamedTuple]) -> Optional[Callable]:
+        """
+        Return the function to execute after each batch on the formated graphQL request result.
+        Args:
+            post_call_options: options given from the call of the GraphQL Query
+        Returns:
+            func: a function that takes the list of formated result from graphQL request and
```
[the remark this answers is not in the document]
Comment (theodu) [reply src/kili/graphql/__init__.py]: outdated with the new commits

File: src/kili/services/copy_project/__init__.py [hunk 2]
```diff
@@ -164,69 +161,66 @@ def _copy_assets(self, from_project_id: str, new_project_id: str):
...
+            "jsonContent",
+            "jsonMetadata",
+        ]
+
+        def download_and_upload_assets(assets):
+            with TemporaryDirectory() as tmp_dir:
+                downloaded_assets = self._download_assets(from_project_id, fields, tmp_dir, assets)
+                return self._upload_assets(new_project_id, downloaded_assets)
+
+        return AssetQuery(self.kili.auth.client)(where, fields, options, download_and_upload_assets)
```
[the remark this answers is not in the document]
Comment (theodu) [reply src/kili/services/copy_project/__init__.py:179]: I agree with you! The post-call method was hard to handle I spent some time trying to get the best architecture. We should discuss this tomorrow during the 1-1.

[the remark this answers is not in the document]
Comment (theodu) [reply src/kili/services/copy_project/__init__.py:179]: I don't merge it yet

**Pull request merged**